OpenSim/Common/Component.cpp

```diff
@@ -1138,6 +1139,21 @@ void Component::updateFromXMLNode(SimTK::Xml::Element& node, int versionNumber)
             }
             
         }
+        if (versionNumber <= 30516) {
+            // Rename xml tags for socket_*_connectee_name to socket_*
+            for (auto iter = node.element_begin();
+                iter != node.element_end();
+                ++iter) {
+                auto tagname = iter->getElementTag();
+                if (std::regex_match(tagname, std::regex("(socket_)(.*)(_connectee_name)"))) {
+                    auto pos = tagname.find("_connectee_name");
+                    if (pos != std::string::npos) {
+                        tagname.replace(pos, 15, "");
```

Comment:
It would be preferable to create an `std::string` variable for `"connectee_name"` and to use `size()` here instead of hardcoding 15.

Reply:
:+1: good call!

Reply:
Done

Reply:
@chrisdembia The problem with inputs as I pointed out earlier is that there are two different tags e.g.
'<input_inputs_connectee_names>' or '<input_transform_connectee_name>' should both be removed losing the fact that one of them is plural?

Reply:
Yes.